### PR TITLE
[ROCm] add ROCm RCCL support

### PR DESCRIPTION
**Pull request by jeffdaily**

### CODE REVIEW

File: tensorflow/core/nccl/nccl_manager.cc [hunk 1]
```diff
@@ -16,15 +16,32 @@ limitations under the License.
...
+#if GOOGLE_CUDA
+#include "tensorflow/core/platform/cuda.h"
+#elif TENSORFLOW_USE_ROCM
+#include "tensorflow/core/platform/rocm.h"
+#endif
 
 namespace tensorflow {
 
+#if GOOGLE_CUDA
+using se::cuda::ScopedActivateExecutorContext;
```
Comment (chsigg) [inline tensorflow/core/nccl/nccl_manager.cc]: Type alias would be preferred over using declarations here and defines below.

**Review by chsigg — DISMISSED** [review]

**Comment by rthadur** [comment]
@jeffdaily can you please resolve conflicts ?

File: tensorflow/core/nccl/nccl_manager.cc (hunk 1 above)
Comment (jeffdaily) [reply tensorflow/core/nccl/nccl_manager.cc]: How about the following?
```C++
#if GOOGLE_CUDA
using ScopedActivateExecutorContext = se::cuda::ScopedActivateExecutorContext;
#elif TENSORFLOW_USE_ROCM
using ScopedActivateExecutorContext = se::rocm::ScopedActivateExecutorContext;
// Local hipify of cuda symbols
using cudaError_t = hipError_t;
using cudaStream_t = hipStream_t;
const auto cudaGetErrorString = hipGetErrorString;
const auto cudaGetDevice = hipGetDevice;
const auto cudaSetDevice = hipSetDevice;
const auto cudaSuccess = hipSuccess;
#endif
```

**Comment by chsigg** [comment]
FYI, I'm working on getting this merged.

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by rthadur** [comment]
Seems auto-merge is not happening but the changes are now committed so we can close this. Thank you for the PR.

**Pull request closed without merging**